spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,305 @@
+require_relative "spec_helper"
+
+describe "Reservation_manager" do
+  let (:res_manager) {
+    Reservation_manager.new
+  }
+  let (:res_one) {
+    res_manager.make_reservation(1, check_in_day: "3rd April 2019", check_out_day: "10th April 2019")
+  }
```

Comment:
Great work with the let syntax! However, remember: let syntax is mostly used for declaring let variables. In a lot of cases, you end up using let as a shortcut to calling methods! I know this because in a lot of cases, you'll have code like `res_one res_two res_three`, which shows me that you're just trying to call it. Instead of using the let syntax, you might want to make helper methods in your tests instead (did you know you could do that? :) )

Writing this is also a good opportunity to change these names:

```ruby
def make_res_one
    return res_manager.make_reservation(1, check_in_day: "3rd April 2019", check_out_day: "10th April 2019")
end
```
To make this work, you'll have to change a few of your tests to conform to this (to something like `res_one = make_res_one`)

I like this style more because then in your tests, some of these lines would make more contextual sense, so you can call `make_res_one make_res_two make_res_three` and it feels more natural

If I were to go all out in refactoring (which would take a lot of time upfront, but may be good!), I would even change the names away from `res_one` to something like `valid_early_res` or something.